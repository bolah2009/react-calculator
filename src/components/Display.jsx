import React from 'react';
import PropTypes from 'prop-types';
import styles from '../css/modules/Display.module.css';

const Display = ({ result }) => (
  <div className={`d-flex col jc-c ${styles.display}`}>{result}</div>
);

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
