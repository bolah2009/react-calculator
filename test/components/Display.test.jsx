import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Display from '../../src/components/Display';

describe('Display component', () => {
  test('should render with default result value', () => {
    const display = renderer.create(<Display />);
    expect(display.toJSON()).toMatchSnapshot();
  });

  test('should render with an assigned result value', () => {
    const display = renderer.create(<Display result="20" />);
    expect(display.toJSON()).toMatchSnapshot();
  });

  test('should render with an assigned result value and current button', () => {
    const display = renderer.create(<Display result="20" currentButton="+" />);
    expect(display.toJSON()).toMatchSnapshot();
  });
});

describe('when result prop is supplied', () => {
  it('renders with the supplied result', () => {
    render(<Display result="500" />);

    const container = screen.getByRole('status');
    expect(container).toHaveTextContent('500');
  });
});

describe('when no result prop is supplied', () => {
  it('renders with a default 0 result', () => {
    render(<Display />);

    const container = screen.getByRole('status');
    expect(container).toHaveTextContent('0');
  });
});

describe('when currentButton prop is supplied', () => {
  it('renders with the supplied currentButton', () => {
    render(<Display currentButton="X" />);

    const container = screen.getByRole('presentation');
    expect(container).toHaveTextContent('X');
  });
});

describe('when no currentButton prop is supplied', () => {
  it('renders no currentButton', () => {
    render(<Display />);

    const container = screen.getByRole('presentation');
    expect(container).toHaveTextContent('');
  });
});
