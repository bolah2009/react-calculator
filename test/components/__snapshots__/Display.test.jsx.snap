// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component should render with an assigned result value 1`] = `
<div
  className="d-flex col jc-c display"
>
  <div
    className="currentButton"
    role="presentation"
  />
  <div
    className="mainDisplay"
    id="display"
    role="status"
  >
    20
  </div>
</div>
`;

exports[`Display component should render with an assigned result value and current button 1`] = `
<div
  className="d-flex col jc-c display"
>
  <div
    className="currentButton"
    role="presentation"
  >
    +
  </div>
  <div
    className="mainDisplay"
    id="display"
    role="status"
  >
    20
  </div>
</div>
`;

exports[`Display component should render with default result value 1`] = `
<div
  className="d-flex col jc-c display"
>
  <div
    className="currentButton"
    role="presentation"
  />
  <div
    className="mainDisplay"
    id="display"
    role="status"
  >
    0
  </div>
</div>
`;
